tests/foreman/longrun/test_capsule_loadbalancer.py

```diff
@@ -0,0 +1,315 @@
+            5. Package installatoin fail
+
+        :CaseLevel: Integration
+
+        :CaseComponent: System
+        """
+        pass
+
+    @pytest.mark.stubbed
+    @tier1
+    def test_remote_execution_on_client(self):
+        """Remote Execution on client
+
+        :id: e911acb4-e153-11ea-9a77-4ceb42ab8dbc
+
+        :Steps:
+            1. Enable foreman-proxy-plugin-remote-execution-ssh on capsules
+            2. Create ansible role and remote job, attach to the host(here client) and run the job
+
+        :expectedresults:
+            1. Ansible job is successful
+            2. Remote job is succesful
+
+        :CaseLevel: Integration
+
+        :CaseComponent: System
+        """
+        pass
+
+    @pytest.mark.stubbed
+    @tier1
+    def test_openscap_reporting(self):
+        """openscap report to be uploaded
+
+        :id: cfba6e54-e158-11ea-9a77-4ceb42ab8dbc
+
+        :Steps:
+            1. Run `Puppet agent -t --noop`
+            2. Run `foreman-scap-client <id>`
+
+        :expectedresults:
+            1. Should create openscap config file
+            2. Should upload reports to sat
+
+        :CaseLevel: Integration
+
+        :CaseComponent: Reporting
+        """
+        pass
+
+    @pytest.mark.stubbed
+    @tier1
+    def test_capsule_3(self):
+        """Tests for capsule_3
+
+        :id: 16fc8490-e15a-11ea-9a77-4ceb42ab8dbc
+
+        :Steps:
+            1. Install capsule_3 and register as "The initial setup[5]".
+            2. Unregister and remove katello-ca-consumer-latest. Re-register the client.
+            3. Try package installation.
+            4. Run `puppet agent -t -v`.
+            5. Capsule_1 and capsule_2 service down.
+            6. Capsule_1 and capsule_2 VM down.
+            7. Capsule_1 and capsule_2 VM up.
+
+        :expectedresults:
+            1. Capsule_3 install successful
+            2. Package installation is successful
+            3. Scap reports upload to satellite should be served through capsule_3
+            4. Request to be severed by capsule_3 when capsule_1 and capsule_2 service/VM are down.
+            5. On starting the services/VM on down capsules, all the capsules including capsule_3
+                should serve the client requests
+
+        :CaseLevel: Integration
+
+        :CaseComponent: System
+        """
+        pass
+
+    @pytest.mark.stubbed
+    @tier1
+    def test_multiclient(self):
+        """Testing multiple clients to test the loadbalancer functionality
+
+        :id: 8b59c69c-e15d-11ea-9a77-4ceb42ab8dbc
+
+        :Steps: Use docker to spin new container client
+
+        :expectedresults: All the clients(from docker) are registered to the capsules.
+
+        :CaseLevel: Integration
+
+        :CaseComponent: System
+        """
+        pass
+
+    @pytest.mark.stubbed
+    @tier1
+    def test_upgrade_the_setup(self):
+        """Upgrading the setup previous GA to current snap
```

Comment:
Same here, this will probably need to somehow tie into CI and `tests/upgrades/` directory and https://github.com/SatelliteQE/satellite6-upgrade repository.

Reply:
The scope of this test was to make sure the certificates are not changed when an upgrade is done in a loadbalancer environment. (as it is necessary to pass `--foreman-proxy-cname loadbalancer.example.com` while running `capsule-certs-generate`). I don't think tying it to CI is necessary.

Reply:
Again, I commented from position that these tests are to be automated soon. Since that's not a plan, this comment is not really relevant.